Automattic/woocommerce-payments
# Dispute Listing: Add smart-date respond by column

Fixes #6702 

#### Changes proposed in this Pull Request

- On dispute list page, rename the 'Respond by' column to 'Response date'.
- If dispute is due within 72 hours, show a smart date, ie "{n} days left" in red with an exclamation mark icon.

ps:
- When I searched for 'Respond by', I found in [dispute details page](https://github.com/Automattic/woocommerce-payments/blob/fix/6702-dispute-listing-smart-dates/client/disputes/info/index.tsx#L27). I don't change this as it's not in the scope.
- [The issue](https://github.com/Automattic/woocommerce-payments/issues/6702) mentions "CSV exports must not be changed (headers or content)". I see that the content of that column does not change in CSV, ie due date is not 'smart' but I can't find a way to make the column header does not change.
- As mentioned p1689145193840379/1689136852.778029-slack-C03LA109BKM, we can utilize `isDueWithin()` from https://github.com/Automattic/woocommerce-payments/pull/6723 but since that PR is not yet merged, this should be done in a separate PR.

#### Testing instructions

* Open dispute list page.
* I mock the response of `/payments/disputes` request by hardcoding it [here](https://github.com/Automattic/woocommerce-payments/blob/fix/6702-dispute-listing-smart-dates/includes/admin/class-wc-rest-payments-disputes-controller.php#L92).

First, I see the response from the network dev tool.
Then, I add this code [here](https://github.com/Automattic/woocommerce-payments/blob/fix/6702-dispute-listing-smart-dates/includes/admin/class-wc-rest-payments-disputes-controller.php#L92).
```php
return json_decode('{
  "data": [
    {
      "wcpay_disputes_cache_id": 27,
      ...
    }
  ]
}', true);
```

So I can change the `due_by` easily.

Test:
- When due date is <= 72 hours from now.
![Screenshot 2023-07-13 at 06 34 42](https://github.com/Automattic/woocommerce-payments/assets/73803630/ce03c212-917e-4b9b-a378-e674ba307e25)


- When due date is <= 24 hours from now.
<img width="164" alt="image" src="https://github.com/Automattic/woocommerce-payments/assets/3147296/c200a532-6595-4b07-93f7-92159417b05a">


- When due date is > 72 hours from now.
![Screenshot 2023-07-13 at 06 35 21](https://github.com/Automattic/woocommerce-payments/assets/73803630/14f4cbe2-8afa-4f28-8ab8-47bb241aa064)


-------------------


**Post merge**
